framework/src/utils/ExtraIDUtils.C

```diff
@@ -0,0 +1,77 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ExtraIDUtils.h"
+#include "libmesh/parallel.h"
+
+std::map<dof_id_type, dof_id_type>
+ExtraIDUtils::ParseExtraIDs(const MeshBase & mesh,
+                            const std::set<SubdomainID> & block_ids,
+                            std::vector<ExtraElementIDName> extra_ids)
+{
+  // check block restriction
+  const bool block_restricted = block_ids.find(Moose::ANY_BLOCK_ID) == block_ids.end();
+  // get element id name of interest in recursive parsing algorithm
+  ExtraElementIDName id_name = extra_ids.back();
+  extra_ids.pop_back();
+  const auto id_index = mesh.get_elem_integer_index(id_name);
+  // create base parsed id set
+  if (extra_ids.empty())
+  {
+    // get set of extra id values;
+    std::set<dof_id_type> ids;
+    for (const auto & elem : mesh.active_local_element_ptr_range())
+    {
+      if (block_restricted && block_ids.find(elem->subdomain_id()) == block_ids.end())
+        continue;
+      auto id = elem->get_extra_integer(id_index);
+      if (!ids.count(id))
+        ids.insert(id);
```

Comment:
```suggestion
      ids.insert(id);
```

No need to check with count - insert will do nothing if it already exists